addons/stock/models/product.py

```diff
@@ -644,6 +670,30 @@ def _compute_quantities_dict(self):
             }
         return prod_available
 
+    def _compute_nbr_moves(self):
+        res = defaultdict(lambda: {'moves_in': 0, 'moves_out': 0})
+        incoming_moves = self.env['stock.move.line'].read_group([('product_id.product_tmpl_id', 'in', self.ids),
+                                                                 ('state', '=', 'done'),
+                                                                 ('picking_code', '=', 'incoming'),
+                                                                 ('date', '>=', fields.Datetime.now() - relativedelta(years=1))],
+                                                                ['product_id'], ['product_id'])
+        outgoing_moves = self.env['stock.move.line'].read_group([('product_id.product_tmpl_id', 'in', self.ids),
+                                                                 ('state', '=', 'done'),
+                                                                 ('picking_code', '=', 'outgoing'),
+                                                                 ('date', '>=', fields.Datetime.now() - relativedelta(years=1))],
+                                                                ['product_id'], ['product_id'])
+        for move in incoming_moves:
+            product = self.env['product.product'].browse([move['product_id'][0]])
+            product_tmpl_id = product.product_tmpl_id.id
+            res[product_tmpl_id]['moves_in'] += int(move['product_id_count'])
+        for move in outgoing_moves:
+            product = self.env['product.product'].browse([move['product_id'][0]])
+            product_tmpl_id = product.product_tmpl_id.id
+            res[product_tmpl_id]['moves_out'] += int(move['product_id_count'])
+        for template in self:
+            template.nbr_moves_in = int(res[template.id]['moves_in'])
+            template.nbr_moves_out = int(res[template.id]['moves_out'])
```

Comment:
Why not a `search_count` ?
```suggestion
        for template in self:
            domain = [
                ('product_id', 'in', template.product_variant_ids.ids),
                ('state', '=', 'done'),
                ('date', '>=', fields.Datetime.now() - relativedelta(years=1)),
            ]
            template.nbr_moves_in = self.env['stock.move'].search_count(
                expression.AND([domain, [('picking_code', '=', 'incoming')]])
            )
            template.nbr_moves_out = self.env['stock.move'].search_count(
                expression.AND([domain, [('picking_code', '=', 'outgoing')]])
            )
```